sw/device/silicon_creator/lib/drivers/alert.c

```diff
@@ -253,3 +257,112 @@ rom_error_t alert_ping_enable(void) {
   SEC_MMIO_WRITE_INCREMENT(/*value=*/2);
   return kErrorOk;
 }
+
+uint32_t alert_config_crc32(void) {
+  uint32_t crc;
+  crc32_init(&crc);
+
+  for (size_t i = 0, offset = ALERT_HANDLER_ALERT_EN_SHADOWED_0_REG_OFFSET;
+       i < ALERT_HANDLER_ALERT_EN_SHADOWED_MULTIREG_COUNT;
+       ++i, offset += sizeof(uint32_t)) {
+    crc32_add32(&crc, abs_mmio_read32(kBase + offset));
+  }
+
+  for (size_t i = 0, offset = ALERT_HANDLER_ALERT_CLASS_SHADOWED_0_REG_OFFSET;
+       i < ALERT_HANDLER_ALERT_CLASS_SHADOWED_MULTIREG_COUNT;
+       ++i, offset += sizeof(uint32_t)) {
+    crc32_add32(&crc, abs_mmio_read32(kBase + offset));
+  }
+
+  for (size_t i = 0, offset = ALERT_HANDLER_LOC_ALERT_REGWEN_0_REG_OFFSET;
+       i < ALERT_HANDLER_LOC_ALERT_REGWEN_MULTIREG_COUNT;
+       ++i, offset += sizeof(uint32_t)) {
+    crc32_add32(&crc, abs_mmio_read32(kBase + offset));
+  }
+
+  for (size_t i = 0, offset = ALERT_HANDLER_LOC_ALERT_EN_SHADOWED_0_REG_OFFSET;
+       i < ALERT_HANDLER_LOC_ALERT_EN_SHADOWED_MULTIREG_COUNT;
+       ++i, offset += sizeof(uint32_t)) {
+    crc32_add32(&crc, abs_mmio_read32(kBase + offset));
+  }
+
+  for (size_t i = 0,
+              offset = ALERT_HANDLER_LOC_ALERT_CLASS_SHADOWED_0_REG_OFFSET;
+       i < ALERT_HANDLER_LOC_ALERT_CLASS_SHADOWED_MULTIREG_COUNT;
+       ++i, offset += sizeof(uint32_t)) {
+    crc32_add32(&crc, abs_mmio_read32(kBase + offset));
```

Comment:
You're right. I was on the fence about this one since what we need here is basically `crc32()` with the context variable, i.e. `crc32(ctx, buf, len)`. I'll try to update that function and use it here.